docs/apache-airflow/faq.rst

```diff
@@ -159,72 +215,194 @@ simple dictionary.
+            dag=dag
+        )
 
-#. Set ``explicit_defaults_for_timestamp = 1`` under the ``mysqld`` section in your ``my.cnf`` file.
-#. Restart the Mysql server.
+This will echo "day={{ ds }}" instead of "day=2020-01-01" for a dagrun with the execution date 2020-01-01 00:00:00.
 
+.. code-block:: python
 
-How to reduce airflow dag scheduling latency in production?
------------------------------------------------------------
+        bo = BashOperator(
+            task_id='my_task',
+            bash_command="echo day={{ ds }}",
+            dag=dag
+        )
 
-Airflow 2 has low DAG scheduling latency out of the box (particularly when compared with Airflow 1.10.x),
-however if you need more throughput you can :ref:`start multiple schedulers<scheduler:ha>`.
+By using the ds macros directly in the template_field, the rendered value results in "day=2020-01-01".
 
-Why next_ds or prev_ds might not contain expected values?
----------------------------------------------------------
+
+Why ``next_ds`` or ``prev_ds`` might not contain expected values?
+------------------------------------------------------------------
 
 - When scheduling DAG, the ``next_ds`` ``next_ds_nodash`` ``prev_ds`` ``prev_ds_nodash`` are calculated using
   ``execution_date`` and ``schedule_interval``. If you set ``schedule_interval`` as ``None`` or ``@once``,
   the ``next_ds``, ``next_ds_nodash``, ``prev_ds``, ``prev_ds_nodash`` values will be set to ``None``.
 - When manually triggering DAG, the schedule will be ignored, and ``prev_ds == next_ds == ds``
 
+
+Task execution interactions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+What does TemplateNotFound mean?
+---------------------------------
+
+TemplateNotFound errors are usually due to misalignment with user expectations when passing path to operator
+that trigger Jinja templating. A common occurrence is with :ref:`BashOperators<howto/operator:BashOperator>`.
+
+Another commonly missed fact is that the files are resolved relative to where the pipeline file lives. You can add
+other directories to the ``template_searchpath`` of the DAG object to allow for other non-relative location.
+
+
+How to trigger tasks based on another task's failure?
+-----------------------------------------------------
+
+For tasks that are related through dependency, you can set the ``trigger_rule`` to ``TriggerRule.ALL_FAILED`` if the
+task execution depends on the failure of ALL its upstream tasks or ``TriggerRule.ONE_FAILED`` for just one of the
+upstream task.
+
+.. code-block:: python
+
+    from airflow.models import DAG
+    from airflow.operators.python import PythonOperator
+    from airflow.utils.trigger_rule import TriggerRule
+
+    parent = PythonOperator(
+        task_id='a',
+        python_callable=lambda x: 1,
+        dag=dag
+    )
+
+    child = PythonOperator(
+        task_id='b',
+        python_callable=lambda: 1,
+        trigger_rule=TriggerRule.ALL_FAILED,
+        dag=dag
+    )
+
+    parent >> child
```

Comment:
lets use Airflow 2 syntax with dag & task decorators.

Reply:
Sounds good! Will convert to task decorator.